src/Fixer/Phpdoc/PhpdocAlignFixer.php

```diff
@@ -306,7 +306,7 @@ private function fixDocBlock(DocBlock $docBlock): void
                     continue;
                 }
 
-                $hasStatic |= '' !== $item['static'];
+                $hasStatic = $hasStatic || '' !== $item['static'];
```

Comment:
was there sth wrong with original approach ?

Reply:
Yes, see the PHPStan errors that are gone with the changes.

Reply:
let me ask differently
what was wrong with original approach ?
(other than PHPStan complaining)

as you can imagine, I wonder why phpstan decided to complain about it, as I see nothing wrong with original code

Reply:
> what was wrong with original approach ?

I don't know what to say, as I was sure that it is more than obvious that we want to have only boolean expressions in logical conditions.

Just to be sure - do you know what will display the below code?
```php
<?php
$x = false;
$x |= true;
var_dump(get_debug_type($x));
```

Reply:
you see? when you want to, you can explain things ;)

Thanks, TIL

next step - avoid need to be asked twice before answering the question, please :thanks:

Reply:
IMHO such reasoning should be put in the PR description or commit message, so the reviewers does not need to figure it out on their own. Such things not necessarily are obvious.

Reply:
sometimes hard to guess what is obvious and what not - different ppl , different things are obvious